run.py

```diff
@@ -102,11 +102,18 @@ def main():
     docker_arch = args.platform or ("linux/amd64/v2" if branch == "9.0" else "linux/amd64")
 
     docker_args = [RUNNER, "run", "-i", "-t",
-                   "-u", "builder",
                    "--platform", docker_arch,
                    ]
     if is_podman(RUNNER):
-        docker_args += ["--userns=keep-id", "--security-opt", "label=disable"]
+        # With podman we use the `--userns` option to map the builder user to
+        # the user on the system.
+        docker_args += [f"--userns=keep-id:uid=1000,gid=1000", "--security-opt", "label=disable"]
```

Comment:
The fact the uid is preserved so entrypoint is entered as `builder` and not as `root` as in the Docker case is likely useful to document.  But is it really a good idea to have such a big difference between the 2 runners? (especially as we want to move more root actions into the entrypoint in the future)